### PR TITLE
Check $strictNullComparison outside of loops

### PR DESCRIPTION
**Pull request by ConnectGrid**
Might be considered a micro-optimization, but for large sheets, it could make a difference in performance. Only check the `$strictNullComparison` condition once, not for every single cell loop.

This is:

- [ ] a bugfix
- [ ] a new feature
- [X] refactoring
- [ ] additional unit tests

Checklist:

- [X] Changes are covered by unit tests
  - [X] Changes are covered by existing unit tests
  - [ ] New unit tests have been added
- [X] Code style is respected
- [X] Commit message explains **why** the change is made (see https://github.com/erlang/otp/wiki/Writing-good-commit-messages)
- [ ] CHANGELOG.md contains a short summary of the change and a link to the pull request if applicable
- [ ] Documentation is updated as necessary

### Why this change is needed?

Instantiating Worksheets from an array are currently less than optimal due to checking a boolean condition on every single cell loop.

### CODE REVIEW

File: src/PhpSpreadsheet/Worksheet/Worksheet.php [hunk 1]
```diff
@@ -2922,23 +2922,30 @@ public function fromArray(array $source, $nullValue = null, $startCell = 'A1', $
...
-            $currentColumn = $startColumn;
-            foreach ($rowData as $cellValue) {
-                if ($strictNullComparison) {
+        if ($strictNullComparison) {
+            foreach ($source as $rowData) {
+                $currentColumn = $startColumn;
+                foreach ($rowData as $cellValue) {
                     if ($cellValue !== $nullValue) {
                         // Set cell value
                         $this->getCell($currentColumn . $startRow)->setValue($cellValue);
```
Comment (MarkBaker) [inline src/PhpSpreadsheet/Worksheet/Worksheet.php:2931]: I've not run any timing tests on your code yet; but when it comes to micro-optimisations, then interpolation is more efficient than the concatenation that we use here:
```php
$this->getCell("{$currentColumn}{$startRow}")->setValue($cellValue);
```

Try running
```php
$callStartTime = microtime(true);

for ($row = 1; $row <= 65535; ++$row) {
    for ($column = 'A'; $column !== 'IV'; ++$column) {
        $cellAddress = $column . $row;
    }
}

$callEndTime = microtime(true);
$concatenationCallTime = $callEndTime - $callStartTime;

echo PHP_EOL;
echo 'Call time for concatenation was ' , sprintf('%.4f', $concatenationCallTime) , ' seconds' , PHP_EOL;


$callStartTime = microtime(true);

for ($row = 1; $row <= 65535; ++$row) {
    for ($column = 'A'; $column !== 'IV'; ++$column) {
        $cellAddress = "{$column}{$row}";
    }
}

$callEndTime = microtime(true);
$interpolationCallTime = $callEndTime - $callStartTime;

echo PHP_EOL;
echo 'Call time for interpolation was ' , sprintf('%.4f', $interpolationCallTime) , ' seconds' , PHP_EOL;
```
to see the difference

And I'm of the growing opinion that interpolation is actually more readable

Comment (oleibman) [reply src/PhpSpreadsheet/Worksheet/Worksheet.php:3157]: Ticket amazingly has no merge conflicts after all this time. I have run Mark's test program, and it now seems to favor concatenation over interpolation. So I will let this proceed as-is, assuming it still passes all tests, of course.

**Comment by oleibman** [comment]
Nothing more to discuss.

**Comment by oleibman** [comment]
Thank you for your contribution.

**Pull request merged**